libminifi/include/io/tls/TLSSocket.h

```diff
@@ -173,19 +173,19 @@ class TLSSocket : public Socket {
    * @param buflen buffer to write
    *
    */
-  int writeData(std::vector<uint8_t> &buf, int buflen);
+  int writeData(const std::vector<uint8_t> &buf, int buflen);
```

Comment:
if we chose ( and we eventually will in further tickets ) to take this memory and do a memmove ( in a different stream -- can discuss later ) we can't do that with the changes. This was intentional and not an oversight.